torch_xla/csrc/tensor.cpp

```diff
@@ -492,28 +492,32 @@ XLATensorPtr XLATensor::Create(
 
 XLATensor::XLATensor(const at::Tensor& tensor,
                      const torch::lazy::BackendDevice& device)
-    : data_(std::make_shared<Data>(tensor, device)) {}
+    : XLATensor(std::make_shared<Data>(tensor, device)) {}
 
 XLATensor::XLATensor(torch::lazy::BackendDataPtr xla_data,
                      c10::optional<at::ScalarType> logical_element_type)
-    : data_(std::make_shared<Data>(xla_data, xla_data->device(),
-                                   logical_element_type)) {}
+    : XLATensor(std::make_shared<Data>(xla_data, xla_data->device(),
+                                       logical_element_type)) {}
 
 XLATensor::XLATensor(torch::lazy::Value ir_value,
                      const torch::lazy::BackendDevice& device,
                      c10::optional<at::ScalarType> logical_element_type)
-    : data_(std::make_shared<Data>(std::move(ir_value), device,
-                                   logical_element_type)) {
+    : XLATensor(std::make_shared<Data>(std::move(ir_value), device,
+                                       logical_element_type)) {
   TryLimitGraphSize();
 }
 
 XLATensor::XLATensor(std::shared_ptr<View> view,
                      const torch::lazy::BackendDevice& device,
                      c10::optional<at::ScalarType> logical_element_type)
-    : data_(std::make_shared<Data>(std::move(view), device,
-                                   logical_element_type)) {}
+    : XLATensor(std::make_shared<Data>(std::move(view), device,
+                                       logical_element_type)) {}
 
-XLATensor::XLATensor(std::shared_ptr<Data> data) : data_(std::move(data)) {}
+XLATensor::XLATensor(std::shared_ptr<Data> data)
+    : data_(std::move(data)),
+      storage_(c10::Storage(
+          {}, 0,
```

Comment:
I am a bit confuse, if we always init `Storage` the same way, how can we tell each tensor's storage is different? (I mean it works, but I am not sure why...)

Reply:
It's a pointer. We compare the memory address instead of the values.

Reply:
If I understand correctly, you are comparing two `c10::DataPtr` which is a pointer to decide if two XLATensor has the same storage?

Reply:
That's correct. The actual value stored in the pointer doesn't matter.